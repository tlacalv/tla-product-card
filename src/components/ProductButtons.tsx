import styles from "../styles/styles.module.css";
import React, { CSSProperties, useCallback, useContext } from "react";
import { ProductContext } from "./ProductCard";

export interface Props {
  className?: string;
  style?: CSSProperties;
}

export const ProductButtons = ({ className, style }: Props) => {
  const { increaseBy, counter, maxCount } = useContext(ProductContext);

  const isMaxReached = useCallback(
    () => !!maxCount && maxCount===counter,
    [counter, maxCount]
  )

  return (
    <div className={`${styles.buttonsContainer} ${className}`} style={style}>
      <button className={styles.buttonMinus} onClick={() => increaseBy(-1)}>
        -
      </button>
      <div className={styles.countLabel}>{counter}</div>
      <button disabled={isMaxReached()} className={`${styles.buttonAdd} ${isMaxReached() && styles.disabled}`} onClick={() => increaseBy(1)}>
        +
      </button>
    </div>
  );
};
