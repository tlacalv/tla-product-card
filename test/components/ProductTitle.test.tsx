import React from 'react'
import renderer from 'react-test-renderer'
import {ProductTitle, ProductCard} from '../../src/components';
import { product1 } from '../data/products';


describe('ProductTitle', () => {
  test('must show the component with the customized title', () => {
    const wrapper = renderer.create(
        <ProductTitle title="Custom Product" />
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  test('must show the component with the product name', () => {
    const wrapper = renderer.create(<ProductCard product={product1}>
      {
        () => (
          <ProductTitle title="Custom Product" />
        )
      }
    </ProductCard>)

    expect(wrapper.toJSON()).toMatchSnapshot();
  })
})