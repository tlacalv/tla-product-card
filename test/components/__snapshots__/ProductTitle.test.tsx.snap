// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle must show the component with the customized title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`ProductTitle must show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Custom Product
  </span>
</div>
`;
